### PR TITLE
refactor(module:collapse): implement new control flow

### PR DESCRIPTION
**Pull request by Nicoss54**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Collapse still use the older control flow

Issue Number: N/A

## What is the new behavior?

Collapse component now use the new control flow

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: components/collapse/demo/accordion.ts [hunk 1]
```diff
@@ -4,9 +4,11 @@ import { Component } from '@angular/core';
   selector: 'nz-demo-collapse-accordion',
   template: `
     <nz-collapse nzAccordion>
-      <nz-collapse-panel *ngFor="let panel of panels" [nzHeader]="panel.name" [nzActive]="panel.active">
-        <p>{{ panel.name }} content</p>
-      </nz-collapse-panel>
+      @for (panel of panels; track panel.name) {
```
Comment (HyperLife1119) [inline components/collapse/demo/accordion.ts:7]: We cannot ensure that panel.name is unique.

Comment (Nicoss54) [reply components/collapse/demo/accordion.ts:7]: At this moment $index is more appropriate or just panel

Comment (HyperLife1119) [reply components/collapse/demo/accordion.ts:7]: Let's use `panel` :)

Comment (Nicoss54) [reply components/collapse/demo/accordion.ts:7]: Will change this today

**Comment by Nicoss54** [comment]
@HyperLife1119 all is done

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**